### PR TITLE
Zero sum transfers fix

### PR DESCRIPTION
**Pull request by georgiyazovaliiski**
**Description**:
This PR fixes an issue in the precompiled `TokenTransfers` functionality.

**Related to**: Precompiles

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/SyntheticTxnFactory.java [hunk 1]
```diff
@@ -86,19 +91,25 @@ public SyntheticTxnFactory() {
 						.setToken(nftExchange.getTokenType())
 						.addNftTransfers(nftExchange.nftTransfer()));
 			}
+			final var tokenTransferListBuilder = TokenTransferList.newBuilder();
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/SyntheticTxnFactory.java:94]: Does this mean that each `TokenTransferWrapper` can have at most one `fungibleTransfer`?

File: hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/SyntheticTxnFactory.java [hunk 2]
```diff
@@ -86,19 +91,25 @@ public SyntheticTxnFactory() {
...
 				}
 				if (fungibleTransfer.receiver != null) {
 					tokenTransferListBuilder.addTransfers(fungibleTransfer.receiverAdjustment());
 				}
-				builder.addTokenTransfers(tokenTransferListBuilder);
 			}
+			builder.addTokenTransfers(tokenTransferListBuilder);
+		}
+
+		final var validity = impliedTransfersMarshal.isPureValidated(builder.build());
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/SyntheticTxnFactory.java:108]: Could we check this instead at the beginning of `TransferPrecompile.run()`?

From the [single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle), the `SyntheticTxnFactory` seems like it should focus on just creating `TransactionBody.Builder`s, but not determining if they are valid.

Comment (stoqnkpL) [reply hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/SyntheticTxnFactory.java]: Check moved to `TransferPrecompile.run()`.

File: hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/SyntheticTxnFactory.java (hunk 1 above)
Comment (stoqnkpL) [reply hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/SyntheticTxnFactory.java]: This should be declared inside the for loop and all fungibleTransfers of the TokenTransferWrapper should be added to the transaction body builder. It is fixed.

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Comment by tinker-michaelj** [comment]
Merge conflicts and issues resolved via https://github.com/hashgraph/hedera-services/pull/2742

**Pull request closed without merging**